libpromises/evalfunction.c

```diff
@@ -248,10 +248,11 @@ static JsonElement* VarRefValueToJson(EvalContext *ctx, const FnCall *fp, const
             convert = JsonArrayCreate(RlistLen(value));
             for (const Rlist *rp = value; rp != NULL; rp = rp->next)
             {
-                if (rp->val.type == RVAL_TYPE_SCALAR &&
-                    strcmp(RlistScalarValue(rp), CF_NULL_VALUE) != 0)
+                char* item = RlistScalarValueSafe(rp);
```

Comment:
Hmmm what if it is an `ilist`, i.e. a list of integers? Then the value is skipped completely! Do we have examples that mix JSON and ilists?